### DISCUSSION TITLE
SD card examples

### DISCUSSION THREAD

**Discussion by hicksan** [olikraus/u8g2#1555] [Ideas]
Might it be possible, please, to have a simpler text-based example of directory listing and file reading to test the U8G2 library with an SD card? These cards are notoriously problematic with SPI busses, especially the cheaper ones, so a test example that were easier to diagnose (text data rather than bitmap) would be helpful.

It would also help, I think, to have something written to screen before the Sd card is initialised, just to establish that the U8G2 bit is working in case the SD.begin causes it to lock. A screen that just remains blank could be a problem with either side. This would be really helpful for people just getting started.

A basic file picker menu would be fantastic. The one with ArduinoMenu has memory bugs and doesn't work, so there is nothing else out there.

**Comment by olikraus** [comment]
U8g2 is just a graphics lib and not a menu library.

**Reply by hicksan** [reply]
I know. But U8G2 handles SD card interaction directly, which no menu system does. A simpler example would really help in making that work for users. I just gave a basic example of a standalone function, drawing on your existing work with the list selector example, which would - by itself - be really useful. That's all. Any text-based example would do. The reading of a binary file is, I think, too complicated as an example because one can't interrogate it with text output to investigate why it isn't working - it either is or isn't - which makes diagnosis really difficult for non-experts. More people will be using this for text anyway and having something to modify is a great starting place. I got mine to work eventually so am only trying to give helpful feedback here. It's such a great library that it would be a pity if others gave up because they couldn't get that bit to work.

**Reply by olikraus** [reply]
I am always happy for feedback, but a file picker is NOT a simple example.

**Reply by hicksan** [reply]
Indeed I am sure you are correct. I meant as compared to diagnosing bitmaps. As I said: just a suggestion. Any text based example would be fine and, I am sure, very much appreciated by your users as a stepping stone. Cannot thank you enough for all your other work anyway. I will let you know if I come up with anything that might help.
